### PR TITLE
Revised span-to-lines conversion to produce an empty vec on DUMMY_SP.

### LINKED ISSUE #68808: cargo check --message-format json regression of error span data on empty main.rs file
**Problem**
`cargo check --message-format json` returns invalid span data, particularly `line_end` and `line_start` are set to 0, though [it is guaranteed to be 1-based](https://github.com/oli-obk/cargo_metadata/blob/master/src/diagnostic.rs#L58-L61).
This happens with a completly empty `main.rs` file.

Tested on the following cargo version:
`cargo 1.42.0-nightly (9d32b7b01 2020-01-26)`

Whilst on the stable version the returned span array is empty:
`cargo 1.41.0 (626f0f40e 2019-12-03)`

**Steps**
1. Clone [this repo](https://github.com/Veetaha/cargo-check-regression-1.42-nightly)
2. Run `rustup default nightly`
3. Run `cargo check --message-format json > check.json`

Now you can see invalid span in `check.json` file.

I've already created and formatted `check-1.41.json` and `check-1.42.json` files with the output of different cargo versions.
Here are the crutial differences [1.41](https://github.com/Veetaha/cargo-check-regression-1.42-nightly/blob/861f9913465a8b880aee1e8e0d6ffa906c3fc637/check-1.41.json#L28) and [1.42](https://github.com/Veetaha/cargo-check-regression-1.42-nightly/blob/861f9913465a8b880aee1e8e0d6ffa906c3fc637/check-1.42.json#L28-L46)

**Context**
[The issue was initially reported at rust-analyzer project repo here.](https://github.com/rust-analyzer/rust-analyzer/issues/2973)

### PR DESCRIPTION
**Pull request by pnkfelix**
This required revising some of the client code to stop relying on the returned set of lines being non-empty.

Fix #68808

### CODE REVIEW

File: src/librustc_span/source_map.rs [hunk 1]
```diff
@@ -545,6 +549,9 @@ impl SourceMap {
         // and to the end of the line. Be careful because the line
         // numbers in Loc are 1-based, so we subtract 1 to get 0-based
         // lines.
+        //
+        // FIXME: now that we handle DUMMY_SP up above, we should consider
+        // aseerting the line numbers here are all indeed 1-based.
```
Comment (Centril) [inline src/librustc_span/source_map.rs]:
```suggestion
        // asserting that the line numbers here are all indeed 1-based.
```

**Comment by estebank** [comment]
r=me after addressing Centril's request

**Pull request merged**